source/server/api_listener_impl.cc

```diff
@@ -31,6 +31,33 @@ void ApiListenerImplBase::SyntheticReadCallbacks::SyntheticConnection::raiseConn
   }
 }
 
+// This class wraps a HttpConnectionManager used as an Http::ApiListener and the associated
+// SyntheticReadCallbacks to ensure that both objects have the same lifetime.
+class HttpApiListener::HttpConnectionManagerWrapper : public Http::ApiListener {
```

Comment:
I'd be inclined to call it an APIListenerWrapper, since it's wrapping the listener first and the HCM implicitly but I'll let @RyanTheOptimist tie break.

Reply:
Yeah, I thought that Alyssa's suggestion made sense so I went ahead and renamed the wrapper class.